src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,120 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+
+/**
+ * <h1> Parser</h1>
+ * The Parser class handles directly the user's command. After parsing the command, the Parser object will call the
+ * relevant methods in response to the command.
+ * <p> A Parser object has Storage, Ui, TaskList and Scanner objects as its attribute</p>
+ */
+public class Parser {
+    private Storage storage;
+    private Ui ui;
+    private TaskList taskList;
+    private Scanner sc;
+
+    /**
+     * Constructor for the Parser class. Takes in Storage, Ui, TaskList and Scanner as its input.
+     * @param storage Storage object specific to this parser
+     * @param ui Ui object specific to this parser
+     * @param taskList TaskList object specific to this parser
+     * @param sc Scanner object specific to this parser
+     */
+    public Parser(Storage storage, Ui ui, TaskList taskList, Scanner sc) {
+        this.storage = storage;
+        this.ui = ui;
+        this.taskList = taskList;
+        this.sc = sc;
+    }
+
+    /**
+     * This method makes sense of the user's command and calls the appropriate methods in response.
+     * @param command This is the user's command
+     * @throws DukeException
+     * @throws IOException
+     */
+    public void parse(String command) throws DukeException, IOException {
```

Comment:
This method is too long. Think of how you can break this down. Also `switch` would be better here instead of `if-else`.